toolkit/scripts/repoquerywrapper.mk

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Contains:
+#	- Tools to query rpms repos in bulk.
+
+.PHONY: pre-repoquerywrapper clean-repoquerywrapper
+
+repoquerywrapper_state_dir = $(REPO_QUERY_DIR)/repoquerywrapper
+repo_urls_file = $(repoquerywrapper_state_dir)/repo_urls.txt
+repoquerywrapper_chroot_dir = $(repoquerywrapper_state_dir)/chroot
+repoquerywrapper_logs_path = $(LOGS_DIR)/repoquerywrapper/repoquerywrapper.log
+
+$(call create_folder,$(repoquerywrapper_state_dir))
+
+clean-repoquerywrapper: clean-repoquerywrapper
+clean: clean-repoquerywrapper
+clean-repoquerywrapper:
+	@echo Verifying no mountpoints present in $(repoquerywrapper_chroot_dir)
+	$(SCRIPTS_DIR)/safeunmount.sh "$(repoquerywrapper_chroot_dir)" && \
+	rm -rf $(repoquerywrapper_state_dir)
+
+# We always want to run the repoquerywrapper script, it will decide if it needs to download anything and update the flag file if
```

Comment:
Need to update this comment, and consider adjusting the make file a bit here if we want to keep the same sort of behaviour.

This is some weird make magic but here goes:
- Consider dep chain A -> B -> C
- Normally, if C file changes, B will rebuild meaning B timestamp is new, which means A rebuilds, etc...
- If C is a `.PHONY:` target, B will ALWAYS rebuild no matter what
- So, if B rebuilds and updates it's timestamps, A will rebuild

So in the naive case here we have `<some dep>` -> `$(STATUS_FLAGS_DIR)/repoquerywrapper.flag`. What if we only want A to rebuild if things actually change? We can't tell without querying the internet.. though, and we need to run the tool to do that. So, add that .PHONY target in `<some dep>` -> `$(STATUS_FLAGS_DIR)/repoquerywrapper.flag` -> `repoquerywrapper_always_run_phony`. We always run "B" now, but what about A? If nothing changed (ie the repo is the same), there is not point in re-running "A".

So, enter the `*.flag` files... The "output" of the tool is `$(QUERY_OUTPUT_FILE)` but we don't use that as our target, instead we use the `.flag` file. Generally the pattern I've used is:
```make
$(STATUS_FLAGS_DIR)/repoquerywrapper.flag:
    my_tool --output "some_file.txt" && \
    if [ ! -f $@ ] || <some_file.txt changed from last time>; then
        touch $@
    fi
```
This means that "A" only updates on the 1st run, or when the results actually change.

Reply:
thx Daniel - I've updated the make file to implement the logic you've described.